src/lib.rs

```diff
@@ -306,6 +306,7 @@ impl<T: Debug + PrimInt + One + Zero> Vob<T> {
     /// assert_eq!(v.get(1), Some(false));
     /// ```
     pub fn push(&mut self, value: bool) {
+        debug_assert_eq!(self.vec.len(), blocks_required::<T>(self.len));
         if self.len % bits_per_block::<T>() == 0 {
             self.vec.push(T::zero());
```

Comment:
this line would break if you do:

```rust
let v = Vob::new();
v.push(false);
v.pop(); // now self.vec.len() == 1, but self.len() == 0
v.push(true); // now self.vec.len() == 2
v.pop() // ==> Some(false)
```

Reply:
Oh, interesting! Can we break this out into its own test-case and commit? [Doesn't have to be a separate PR.]

Reply:
The doctest for `push` already triggered the `debug_assert_eq!`, but I've still added the testcase.